addons/pos_stripe/models/pos_payment_method.py

```diff
@@ -0,0 +1,92 @@
+
+_logger = logging.getLogger(__name__)
+TIMEOUT = 10
+
+class PosPaymentMethod(models.Model):
+    _inherit = 'pos.payment.method'
+
+    def _get_payment_terminal_selection(self):
+        return super()._get_payment_terminal_selection() + [('stripe', 'Stripe')]
+
+    # Stripe
+    stripe_serial_number = fields.Char(help='[Serial number of the stripe terminal], for example: WSC513105011295', copy=False)
+
+    @api.constrains('stripe_serial_number')
+    def _check_stripe_serial_number(self):
+        if not self.sudo()._get_stripe_secret_key():
+            raise ValidationError(_('Complete the Stripe onboarding.'))
+
+        for payment_method in self:
+            if not payment_method.stripe_serial_number:
+                continue
+            existing_payment_method = self.search([('id', '!=', payment_method.id),
+                                                   ('stripe_serial_number', '=', payment_method.stripe_serial_number)],
+                                                  limit=1)
+            if existing_payment_method:
+                raise ValidationError(_('Terminal %s is already used on payment method %s.',\
+                     payment_method.stripe_serial_number, existing_payment_method.display_name))
+
+
+    @api.model
+    def _get_stripe_secret_key(self):
+        stripe_secret_key = self.env['payment.acquirer'].search([('provider', '=', 'stripe')], limit=1).stripe_secret_key
+
+        if not stripe_secret_key:
+            raise ValidationError(_('Complete the Stripe onboarding.'))
+
+        return stripe_secret_key
+
+    @api.model
+    def stripe_connection_token(self):
+        endpoint = 'https://api.stripe.com/v1/terminal/connection_tokens'
+
+        try:
+            req = requests.post(endpoint, auth=(self.sudo()._get_stripe_secret_key(), ''), timeout=TIMEOUT)
+        except requests.exceptions.RequestException:
+            raise UserError(_("There are some issues between us and Stripe, try again later."))
+
+        if req.status_code is not requests.codes.ok:
+            raise UserError(_("There are some issues to contact Stripe."))
+
+        return req.json()
+
+    def stripe_payment_intent(self, amount):
+        # For Terminal payments, the 'payment_method_types' parameter must include
+        # 'card_present' and the 'capture_method' must be set to 'manual'
+        endpoint = 'https://api.stripe.com/v1/payment_intents'
+        currency = self.journal_id.currency_id or self.company_id.currency_id
+
+        try:
+            data = ("currency=%s&amount=%s&payment_method_types[]=card_present&capture_method=manual") % \
+                (currency.name, int(amount/currency.rounding))
+            req = requests.post(endpoint, data=data, auth=(self.sudo()._get_stripe_secret_key(), ''), timeout=TIMEOUT)
+        except requests.exceptions.RequestException:
+            raise UserError(_("There are some issues between us and Stripe, try again later."))
```

Comment:
thanks for removing the "Odoo" mentions, which I think were there! 🙏

You can generally avoid such awkwardness by removing the (pro)nouns that refer to our software entirely. The other error message below is a good example.